users/postgres/users.go

```diff
@@ -163,6 +171,14 @@ func (ur userRepository) RetrieveAll(ctx context.Context, offset, limit uint64,
 		emq = fmt.Sprintf("WHERE %s AND %s", eq, mq)
 	}
 
+	if len(ids) > 0 {
+		if len(emq) == 0 {
+			emq = fmt.Sprintf("WHERE id IN ('%s')", strings.Join(ids, ",'"))
+		} else {
```

Comment:
Please avoid using `else`  whenever you can. Consider simplifying by using something like this:
```go
...
var query []string
if eq != "" {
	query = append(query, eq)
}
if mq != "" {
	query = append(query, mq)
}
if len(ids) > 0 {
	query = append(query, fmt.Sprintf("id IN ('%s')", strings.Join(ids, ",'")))
}
if len(query) > 0 {
	emq = fmt.Sprintf(" WHERE %s", strings.Join(query, " AND "))
}
...
```